### ISSUE TITLE
GSoC 2018 Project Proposal Draft Urgent Feedback

### ISSUE THREAD

**Issue by RohitLearner** [FreeUKGen/FreeUKRegProductIssues#1055]
Please provide necessary feedbacks and required changes in my Project Proposal  Draft for this late spring.  It is particularly inside and out in nature and generously read every one of the 19 pages. 
[First Draft of proposal idea.pdf](https://github.com/FreeUKGen/FreeUKRegProductIssues/files/1828555/First.Draft.of.proposal.idea.pdf)
or Google Doc Edit Link
https://docs.google.com/document/d/1wqSH4nX1kq_ZK8IZKoQtYHJfAqrNyltGYkj1egTsEuA/edit?usp=sharing

**Comment by PatReynolds** [comment]
Thanks @RohitLearner I have had a quick look and nothing seems obviously lacking - we are meeting tomorrow and will discuss and feed back to you.

**Comment by RohitLearner** [comment]
Much appreciated, @PatReynolds for the reply. Anticipating the nitty gritty criticism.

**Comment by RohitLearner** [comment]
@PatReynolds I have shared the final draft from my GSoC account for the feedback; please give the required comment in the remark of the draft.

**Comment by RohitLearner** [comment]
@PatReynolds A debt of gratitude is in order for your important feedback; since the criticism was excessively near the due time; I was not able to make the corrections. Despite the fact that the reference list and my own commitment to the proposal are specified in this comment.

Reference for Proposal:
1. Tracing Genealogical Data with TimeNets by Nam Wook Kim, Stuart K. Card, Jeffrey Heer
2. Visualizing large genealogies with timelines by Daniyar Mukaliyev (Master’s Thesis)
3. Family Tree Visualization by Kerstin Keller, Prahalika Reddy, Shimul Sachdeva

My Contribution & Future work on the existing proposal : 

1.	One of them happens as a result of intermarriages. Despite the fact that intermarriages are not basic in reality families the measurable examination proposes that all individuals alive today have a typical ancestor (not really exceptional), inferring that all individuals alive are blood descendants. 
In our model when I manufactured the tree of ancestors, I monitored individuals as of now added to the tree and did not allow duplicates. That issue needs examination and a superior solution on the most proficient method to deal with it without losing essential data and still have the capacity to demonstrate these unpredictable connections. 

2.	The missing worldly data. As the position and visual qualities, for example, length of help totally relies upon the dates of birth and passing of individuals, the nonappearance of even one of these parameters makes it difficult to imagine the lifelines. 
As the last work, a superior missing birth and passing dates estimation calculation should be executed. 
3.	Dynamically include nodes and edges: Right now, the user can't progressively add nodes and edges to the tree. In the event that the user needs another node added to the tree, he should add it to the information XML record and reload the tree. 
4.	Add pictures to every node. Moreover, a family tree could incredibly enhance by having pictures for every individual. 
5.	Lay out nodes geologically
